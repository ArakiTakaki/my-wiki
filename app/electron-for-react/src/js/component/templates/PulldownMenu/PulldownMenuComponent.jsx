import React from 'react';
import PropTypes from 'prop-types';
import s from './PulldownMenuStyle.scss';
import classNames from 'classnames';

class PulldownMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: this.props.isOpen || false
    };
  }
  render() {
    const pulldownContentClasses = classNames({
      [s['pulldown-content']]: true,
      [s['-open']]: this.state.isOpen
    });
    return (
      <div>
        <div>
          title <span className={s['-open']} />
        </div>
        <div className={pulldownContentClasses}>{this.props.children}</div>
      </div>
    );
  }
}

PulldownMenu.propTypes = {
  //TODO props: PropTypes.string.isRequired
  children: PropTypes.node,
  isOpen: PropTypes.bool
};

export default PulldownMenu;
