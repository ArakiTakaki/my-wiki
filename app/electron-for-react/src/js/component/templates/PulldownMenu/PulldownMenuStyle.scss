// test

.pulldown-title {
  display: flex;

  & > .-open {
    display: inline-block;
    width: 10px;
    height: 10px;
    border-right: 2px;
    border-bottom: 2px;
    transform: rotate(45deg);
  }

  & > .-close {
    display: inline-block;
    width: 10px;
    height: 10px;
    border-right: 2px;
    border-bottom: 2px;
    transform: rotate(45deg);
  }
}

.pulldown-content {
  display: flex;
  max-height: 0;
  transition: $basic-transition;

  & .-open {
    max-height: 100%;
  }
}
