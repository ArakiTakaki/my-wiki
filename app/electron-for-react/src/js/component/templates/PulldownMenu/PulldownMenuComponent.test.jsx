import React from 'react';
import PulldownMenu from './PulldownMenuComponent';
import renderer from 'react-test-renderer';
// react16.8で使用できないので一旦保留
// import { shallow } from 'enzyme';
// import { configure } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// configure({ adapter: new Adapter() });

// snapshot test
test('PulldownMenu SnapShot', () => {
  let component = null;
  component = renderer.create(<PulldownMenu value="test" />).toJSON();
  expect(component).toMatchSnapshot();
});

/*
test('PulldownMenu enzyme', () => {
  const wrapper = shallow(<PulldownMenu />);
  // TODO コンポーネントのテストを記載する
  // expect(wrapper.text()).toBe('InputType:test');
});
*/
