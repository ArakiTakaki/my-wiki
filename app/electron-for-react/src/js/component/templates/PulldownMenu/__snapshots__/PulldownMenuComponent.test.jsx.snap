// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PulldownMenu SnapShot 1`] = `
<div>
  <div>
    title 
    <span
      className="-open"
    />
  </div>
  <div
    className="pulldown-content"
  />
</div>
`;
